lib/rules/jsx-quotes.js

```diff
@@ -19,11 +21,19 @@ var astUtils = require("../ast-utils");
 var QUOTE_SETTINGS = {
     "prefer-double": {
         quote: "\"",
-        description: "singlequote"
+        alternateQuote: "'",
+        description: "singlequote",
+        convert: function(str) {
+            return toDoubleQuotes(str);
```

Comment:
try `x=<a atr='x &quot; &#39; &#34; &amp; y' />` in babel.js repl https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=true&spec=true&code=x%3D%3Ca%20atr%3D'x%20%26quot%3B%20%26%2339%3B%20%26%2334%3B%20%26amp%3B%20y'%20%2F%3E